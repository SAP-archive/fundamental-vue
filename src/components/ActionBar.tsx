import { Slot, Component, DefaultSlot, Prop, Base } from '@/core';

interface Props {
  title: string;
  description?: string | null;
}

@Component('ActionBar')
@DefaultSlot('custom action buttons')
@Slot('back', 'custom back button')
export class ActionBar extends Base<Props> {
  @Prop('page title', { required: true, type: String })
  public title!: string;

  @Prop('action bar description', { default: null, type: String })
  public description!: string | null;

  public render() {
    const actions = this.$slots.default;
    const hasActions = !!actions;
    const back = this.$slots.back;
    const hasBack = !!back;
    const title = this.title;
    const description = this.description;

    return (
      <div class='fd-action-bar'>
        {hasBack &&
          <div class='fd-action-bar__back'>
            {back}
          </div>
        }
        <div class='fd-action-bar__header'>
          <h1 class='fd-action-bar__title'>{title}</h1>
          {!!description &&
            <p class='fdfd-action-bar__description'>{description}</p>
          }
        </div>
        {hasActions &&
          <div class='fd-action-bar__actions'>
            {actions}
          </div>
        }
      </div>
    );
  }
}
