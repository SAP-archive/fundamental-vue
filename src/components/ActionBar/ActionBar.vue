<template>
  <div class="fd-action-bar">
    <div v-if="hasBack" class="fd-action-bar__back">
      <!-- Custom back-button. By default nothing is rednered. -->
      <slot name="back" />
    </div>
    <div class="fd-action-bar__header">
      <h1 class="fd-action-bar__title">
        <!-- Custom title. You can either provide a custom title-slot or set the title-prop which is used if you don't provide a title-slot. -->
        <slot name="title">
          {{ title }}
        </slot>
      </h1>
      <p v-if="hasDescription" class="fd-action-bar__description">
        <!-- Custom description. You either can provide a custom description-slot or set the description-prop which is used if you don't provide a description-slot. -->
        <slot name="description">
          {{ description }}
        </slot>
      </p>
    </div>
    <div v-if="hasActions" class="fd-action-bar__actions">
      <!-- Custom actions. You should use this slot to add fd-button's. -->
      <slot />
    </div>
  </div>
</template>

<script>
export default {
  name: "FdActionBar",
  props: {
    title: { type: String, required: false },
    description: { type: String, default: null }
  },
  computed: {
    hasDescription() {
      return this.$slots.description != null || this.description != null;
    },
    hasActions() {
      return this.$slots.default != null;
    },
    hasBack() {
      return this.$slots.back != null;
    }
  }
};
</script>
