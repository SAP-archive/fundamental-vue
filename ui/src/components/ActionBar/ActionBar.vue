<template>
  <div class="fd-action-bar">
    <div v-if="hasBack" class="fd-action-bar__back">
      <slot name="back" />
    </div>
    <div class="fd-action-bar__header">
      <h1 class="fd-action-bar__title">
        <slot name="title">
          {{ title }}
        </slot>
      </h1>
      <p v-if="hasDescription" class="fd-action-bar__description">
        <slot name="description">
          {{ description }}
        </slot>
      </p>
    </div>
    <div v-if="hasActions" class="fd-action-bar__actions">
      <slot />
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({
  name: "FdActionBar",
  props: {
    title: { type: String, required: false },
    description: { type: String, default: null }
  },
  computed: {
    hasDescription(): boolean {
      return this.$slots.description != null || this.description != null;
    },
    hasActions(): boolean {
      return this.$slots.default != null;
    },
    hasBack(): boolean {
      return this.$slots.back != null;
    }
  }
});
</script>
