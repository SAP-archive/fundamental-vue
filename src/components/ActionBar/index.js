import ActionBar from "./ActionBar.vue";
import { pluginify } from "./../../util";
export default pluginify(ActionBar);
export { ActionBar };
