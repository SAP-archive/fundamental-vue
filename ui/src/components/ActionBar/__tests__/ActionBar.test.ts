import { mount } from "@vue/test-utils";
import ActionBar from "../ActionBar.vue";

describe("ActionBar", () => {
  it("renders correctly", () => {
    const wrapper = mount(ActionBar, {
      propsData: {
        title: "Page Title",
        description: "Action bar descrtiption"
      },
      slots: {
        back: "Back",
        default: "Actions"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders without description", () => {
    const wrapper = mount(ActionBar, {
      propsData: {
        title: "Page Title"
      },
      slots: {
        back: "Back",
        default: "Actions"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find(".fd-action-bar__description").exists()).toBe(false);
  });
});
