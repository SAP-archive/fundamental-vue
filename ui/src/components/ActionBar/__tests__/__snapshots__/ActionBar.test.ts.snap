// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar Description as slot renders correctly 1`] = `
<div
  class="fd-action-bar"
>
  <!---->
   
  <div
    class="fd-action-bar__header"
  >
    <h1
      class="fd-action-bar__title"
    >
      
        
      
    </h1>
     
    <p
      class="fd-action-bar__description"
    >
      <div>
        Page Description (slot)
      </div>
    </p>
  </div>
   
  <!---->
</div>
`;

exports[`ActionBar Title as slot renders correctly 1`] = `
<div
  class="fd-action-bar"
>
  <!---->
   
  <div
    class="fd-action-bar__header"
  >
    <h1
      class="fd-action-bar__title"
    >
      <div>
        Page Title (slot)
      </div>
    </h1>
     
    <!---->
  </div>
   
  <!---->
</div>
`;

exports[`ActionBar Without description renders correctly 1`] = `
<div
  class="fd-action-bar"
>
  <div
    class="fd-action-bar__back"
  >
    Back
  </div>
   
  <div
    class="fd-action-bar__header"
  >
    <h1
      class="fd-action-bar__title"
    >
      
        Page Title
      
    </h1>
     
    <!---->
  </div>
   
  <div
    class="fd-action-bar__actions"
  >
    Actions
  </div>
</div>
`;

exports[`ActionBar renders correctly 1`] = `
<div
  class="fd-action-bar"
>
  <div
    class="fd-action-bar__back"
  >
    Back
  </div>
   
  <div
    class="fd-action-bar__header"
  >
    <h1
      class="fd-action-bar__title"
    >
      
        Page Title
      
    </h1>
     
    <p
      class="fd-action-bar__description"
    >
      
        Action bar description
      
    </p>
  </div>
   
  <div
    class="fd-action-bar__actions"
  >
    Actions
  </div>
</div>
`;
